dot/network/utils.go

```diff
@@ -217,7 +217,7 @@ func readStream(stream libp2pnetwork.Stream, bufPointer *[]byte, maxSize uint64)
 	}
 
 	tot = 0
-	for i := 0; i < maxReads; i++ {
+	for {
```

Comment:
What if we do something like:

```suggestion
	for tot < int(length) {
```

Letting the `if tot != int(length)` check and we can prevent creating a new error type